src/statistic/update.py

```diff
@@ -273,53 +269,54 @@ def get_architecture_stats(self):
             pipeline_match={'processed_analysis.cpu_architecture.summary': {'$exists': True, '$not': {'$size': 0}}},
             additional_projection={'processed_analysis.cpu_architecture.summary': 1}
         )
-        result = [self._shorten_architecture_string(self._find_most_frequent_architecture(list(itertools.chain.from_iterable(item['architecture']))))
-                  for item in self.db.file_objects.aggregate(aggregation_pipeline)]
+        query_result = self.db.file_objects.aggregate(aggregation_pipeline, allowDiskUse=True)
+        result = [
+            self._shorten_architecture_string(self._find_most_frequent_architecture(list(itertools.chain.from_iterable(item['architecture']))))
+            for item in query_result
```

Comment:
Why is the query_result not inline here anymore?

Reply:
I just thought this might be a bit more readable but I can change it back if you wish